examples/emoji-color-cairo.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#  pycairo/cairocffi-based emoji-color example - Copyright 2017 Hin-Tak Leung
+#  Distributed under the terms of the new BSD license.
+#
+#  This script demonstrates overlapping emojis.
+#
+
+import freetype
+import numpy as np
+from PIL import Image
+
+from cairo import ImageSurface, FORMAT_ARGB32, Context
+
+face = freetype.Face("Apple Color Emoji.ttf")
+# Not all char sizes are valid for emoji fonts;
+# Google's NotoColorEmoji only accept size 109 to get 136x128 bitmaps
+face.set_char_size( 160*64 )
+face.load_char('😀', freetype.FT_LOAD_COLOR)
+
+bitmap = face.glyph.bitmap
+width = face.glyph.bitmap.width
+rows = face.glyph.bitmap.rows
+
+# The line below depends on this assumption. Check.
+if ( face.glyph.bitmap.pitch != width * 4 ):
+    raise RuntimeError('pitch != width * 4 for color bitmap: Please report this.')
+bitmap = np.array(bitmap.buffer, dtype=np.uint8).reshape((bitmap.rows,bitmap.width,4))
+
+I = ImageSurface(FORMAT_ARGB32, width, rows)
+ndI = np.ndarray(shape=(rows,width), buffer=I.get_data(),
+                 dtype=np.uint32, order='C',
+                 strides=[I.get_stride(), 4])
+
+# Although both are 32-bit, cairo is host-order while
+# freetype is small endian.
+ndI[:,:] = bitmap[:,:,3] * 16777216 + bitmap[:,:,2] * 65536 + bitmap[:,:,1] * 256 + bitmap[:,:,0]
```

Comment:
What reshape are you talking about exactly ? The suggestion uses numpy broadcasting  and a simple sum over the last axis then copy the result inplace in the array, taking order and strides into account. Anyway, no need to answer, we can keep the line as it is currently.

However, I cannot test it (yet) because `ImageSurface.get_data()` has been introduced in cairo 1.2. You should check for version at the start of each example to tell user he needs to upgrade (else it is difficult to identify the problem).